config/v1/types_infrastructure.go

```diff
@@ -53,28 +63,43 @@ type InfrastructureStatus struct {
 type PlatformType string
 
 const (
-	// AWSPlatform represents Amazon Web Services infrastructure.
-	AWSPlatform PlatformType = "AWS"
+	// AWSPlatformType represents Amazon Web Services infrastructure.
+	AWSPlatformType PlatformType = "AWS"
 
-	// AzurePlatform represents Microsoft Azure infrastructure.
-	AzurePlatform PlatformType = "Azure"
+	// AzurePlatformType represents Microsoft Azure infrastructure.
+	AzurePlatformType PlatformType = "Azure"
 
-	// GCPPlatform represents Google Cloud Platform infrastructure.
-	GCPPlatform PlatformType = "GCP"
+	// GCPPlatformType represents Google Cloud Platform infrastructure.
+	GCPPlatformType PlatformType = "GCP"
 
-	// LibvirtPlatform represents libvirt infrastructure.
-	LibvirtPlatform PlatformType = "Libvirt"
+	// LibvirtPlatformType represents libvirt infrastructure.
+	LibvirtPlatformType PlatformType = "Libvirt"
 
-	// OpenStackPlatform represents OpenStack infrastructure.
-	OpenStackPlatform PlatformType = "OpenStack"
+	// OpenStackPlatformType represents OpenStack infrastructure.
+	OpenStackPlatformType PlatformType = "OpenStack"
 
-	// NonePlatform means there is no infrastructure provider.
-	NonePlatform PlatformType = "None"
+	// NonePlatformType means there is no infrastructure provider.
+	NonePlatformType PlatformType = "None"
 
-	// VSpherePlatform represents VMWare vSphere infrastructure.
-	VSpherePlatform PlatformType = "VSphere"
+	// VSpherePlatformType represents VMWare vSphere infrastructure.
+	VSpherePlatformType PlatformType = "VSphere"
 )
 
+// PlatformStatus holds the current status specific to the underlying infrastructure provider
+// of the current cluster. Since these are used at status-level for the underlying cluster, it
+// is supposed that only one of the status structs is set.
+type PlatformStatus struct {
+	// AWSPlatformStatus contains settings specific to the Amazon Web Services infrastructure provider.
+	// +optional
+	AWSPlatformStatus *AWSPlatformStatus `json:"awsPlatformStatus,omitempty"`
+}
+
+// AWSPlatformStatus holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformStatus struct {
+	// tags holds key and value pairs that are set as tags in the AWS resources created by the cluster.
+	Tags map[string]string `json:"tags,omitempty"`
```

Comment:
does this actually extend to loadbalancers, volumes, or machines created by machine api?

Reply:
for in-tree aws cloud provider, it uses the cluster id here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/aws.go#L548

Reply:
I think we can say that it should after this merges. I can notify those teams to make sure they're respecting it, if possible. 

Machine API already has this info on each MachineSet, so user created ones wouldn't match unless the user thought to add the same tags.

I'm not sure how ingress/registry/storage are tagging but it would be nice to get everyone on same page. (kubernetes.io/cluster/dgoodwin-dev-6vzj7=owned)

I believe the infrastructureName being added above should match the kube one you link to.

Reply:
> does this actually extend to loadbalancers, volumes, or machines created by machine api?

the installer currently tags all the machine/machinesets, loadbalancers with these tags on aws
eg: https://github.com/openshift/installer/blob/5455c4ac317900208a92983e35e6b87fc74fd354/pkg/asset/machines/aws/machines.go#L114-L129

> for in-tree aws cloud provider, it uses the cluster id here:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/aws.go#L548

These are the tags important to openshift on AWS, all controllers must add these and then add more if it likes.

Reply:
this is a bit chicken/egg i guess.

the in-tree aws cloud provider from kube only applies the tags it knows about here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/tags.go

which right now is of the form:
kubernetes.io/cluster/<cluster-id>=owned | shared

which we are getting assigned to entities by a bit of luck because of how we pre-tag ec2 instances.  on startup, kube cloud provider sees its aws, sees no cloud.conf file to pull the cluster id from, and then falls back to reading the ec2 instance for its tags.

https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/aws.go#L1256

i would prefer that we update the language a bit here to reflect that the full set of tags enumerated may not be used by all entities unless we are saying the only tag we will set here.  what will we set in this map?  do we care about both key names and values?

Reply:
or to put another way, why do we need any tag beyond kubernetes.io/cluster/cluster-id ?

Reply:
I think that's all we need to function, but we also support user specified tags from the install config: https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go#L11 That may be why this added here.

Reply:
why do we let users specify addtiional tags?  what is the use case?

Reply:
cc @wking @abhinavdahiya ^^

Reply:
we can only guarantee one global tag is applied given the state of the project.

kubernetes.io/cluster/cluster-id

the api is not implementable as proposed.

Reply:
We let them specify additional tags for whatever they want ;).  Maybe for billing reports, etc.  We also use this for `expirationDate` to backstop resource reaping, openshift/release#1103

Reply:
If it's holding up this PR, I'm fine adding language about anything beyond `kubernetes.io/cluster/...` being best-effort.

Reply:
For our purposes it was really the InfrastructureName we need as that should be the basis for the KubernetesClusterID, the tag can be inferred. 

If we drop specification of arbitrary tags can we get this through @derekwaynecarr ?

Reply:
> the tag can be inferred.
I use be wary of operator making same decision independently.

Reply:
a best-effort API is hard to understand and enforce.

Reply:
@abhinavdahiya 

> > the tag can be inferred.
> 
> I would be cautious of operators making same decision independently.

How can we make this explicit then?

InfrastructureTag:
  Key: kubernetes.io/cluster/infraName
  Value: owned

InfrastructureTag: kubernetes.io/cluster/infraName=owned (string)

My vote would be just to leave it implicit, if you need to tag cloud resources we expect you to apply the standard kube tag using the InfrastructureName as the Kube cluster-id.